[WIP] Document All The Things

jeffpaul opened the pull request:
- [X] add repo description
- [ ] add setup details
- [ ] add PHPCS server usage details
- [ ] add WPCS usage details
- [ ] add PHP Compatibility usage details
- [ ] add API usage details
- [X] expand Google Lighthouse reporting usage details
- [X] add contributing details
- [X] add table of contents
- [X] pull in content from xwp/wp-tide/readme.md
- [X] add code of conduct
- [ ] update email in code of conduct + contributing
- [X] add issue template
- [x] add pr template

jeffpaul committed:
add documentation to readme.md

Initial draft of documentation, more to come

valendesigns commented on README.md:
```diff
@@ -2,8 +2,45 @@
 
 [![Build Status](https://travis-ci.org/xwp/go-tide.svg?branch=develop)](https://travis-ci.org/xwp/go-tide) [![Coverage Status](https://coveralls.io/repos/github/xwp/go-tide/badge.svg?branch=develop)](https://coveralls.io/github/xwp/go-tide?branch=develop) [![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)
 
+Tide services responsible for polling the WordPress.org APIs for themes and plugins to audit ("Sync Server"), for running audits and sending the results to the Tide API service ("Audit Server"), and for running Google Lighthouse audits against themes in the WordPress.org repository ("Lighthouse Server").
```
Possible changes:
> Tide services are responsible for the following. The Sync Server polls the WordPress.org API's for themes and plugins to process and writes them to a queue. The PHPCS Server reads messages from a queue and runs reports against both plugins and themes, then sends the results back to the Tide API. The Lighthouse Server also reads messages from a queue and runs Google Lighthouse reports against only the themes and send those results back to the Tide API.

valendesigns commented on README.md:
```diff
@@ -2,8 +2,45 @@
+
+### Usage
+
+#### WPCS Auditing
```
I think we should start to steer away from the term `audit` or `auditing`. Also, I would suggest these title changes:
```
### Usage
@todo

### PHPCS Server
@todo

#### WPCS
@todo

#### PHP Compatibility
@todo

### Lighthouse Server
This section outlines ...
```

rheinardkorf replied:
@valendesigns would you suggest renaming the `audit` package in `wptide/pkg`?  Perhaps `process` ?

jeffpaul committed:
remove audit* + update section titles in readme.md

jeffpaul committed:
update repo description

jeffpaul committed:
adds Lighthouse Server components

jeffpaul committed:
add section formatting to Lighthouse Server notes

jeffpaul committed:
add Lighthouse Server process notes

valendesigns committed:
Merge branch 'develop' into documentation-updates

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+The following outlines the components added to Tide in order to integrate Lighthouse in overall Tide auditing of WordPress themes.
+1. Docker Container
+   - Lighthouse CLI
+   - Lighthouse Audit Server binary (lh-server)
```
Correction: `Lighthouse Server binary (lighthouse-server)`

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+   - Lighthouse Audit Server binary (lh-server)
+   **Note:** Uses an Alpine Image with working Chromium version. Produces consistent results with Lighthouse Chrome extension. Allows reduced image size of 432MB.
+2. Go Lighthouse Server Source Code
+   - `cmd/lh-server` for binary
```
Correction: `cmd/lighthouse-server` for binary

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+   **Note:** Uses an Alpine Image with working Chromium version. Produces consistent results with Lighthouse Chrome extension. Allows reduced image size of 432MB.
+2. Go Lighthouse Server Source Code
+   - `cmd/lh-server` for binary
+   - `src` for packages
```
Correction: There is no `src` directory.

jeffpaul replied:
@rheinardkorf is there an update for packages that can be applied to the documentation here?

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+   - `cmd/lh-server` for binary
+   - `src` for packages
+   - `vendor` for imported packages
+     - [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go)
```
Correction: We only import two packages.
`github.com/nanobox-io/golang-scribble`
`github.com/wptide/pkg`

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+   - `vendor` for imported packages
+     - [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go)
+     - [github.com/hhatto/gocloc](https://github.com/hhatto/gocloc)
+   - Running Tide cluster
```
Why is this this item here?

jeffpaul replied:
@rheinardkorf this is from the source gdoc you wrote. any recollection what this bullet was for?

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+
+#### Process
+The following demonstrates how a WordPress theme is run through a Lighthouse audit and has its results stored and returned via the Tide API.
+1. `tide-cluster`
```
There is no such thing as a `tide-cluster`

jeffpaul replied:
@rheinardkorf is there an update for the documentation that can be applied here?

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+1. `tide-cluster`
+   - Starts all Tide services and listens...
+@todo add screenshot
+2. `lh-server`
```
This is `lighthouse-server`

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+   - downloads theme and calculates checksum
+@todo add screenshot
+   - runs source through `gocloc` to get code information
+   - scans source for Theme header information (required for Lighthouse Audit)
```
Change `Lighthouse Audit` to `Lighthouse Report`.

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+   - runs source through `gocloc` to get code information
+   - scans source for Theme header information (required for Lighthouse Audit)
+@todo add screenshot
+   - runs theme through `lighthouse-cli` at `https://wp-themes.com/<theme-slug>` and keeps polling for next job.
```
The cli command is just called `lighthouse`.

valendesigns commented on README.md:
```diff
@@ -2,8 +2,85 @@
+   - runs theme through `lighthouse-cli` at `https://wp-themes.com/<theme-slug>` and keeps polling for next job.
+@todo add screenshot
+   - saves full report to a file store (S3)
+@todo add screenshot
```
Missing a step where it takes the full report and generates the detailed report.

jeffpaul replied:
@valendesigns @rheinardkorf are there any commands/screenshots that help highlight that added step?

jeffpaul committed:
add screenshot to LH server process section

jeffpaul committed:
Merge branch 'documentation-updates' of https://github.com/xwp/go-tide into documentation-updates

jeffpaul committed:
readme.md updates from Derek's review

jeffpaul committed:
adds remaining screenshots to lh server process section

jeffpaul committed:
adds docs to Lighthouse Results in Tide API section

jeffpaul committed:
add contributing details

jeffpaul committed:
add table of contents

valendesigns commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,25 @@
+
+- Checkout a topic branch from `develop` and merge back against `develop`.
+    - If you are not familiar with branching please read [_A successful Git branching model_](http://nvie.com/posts/a-successful-git-branching-model/) before you go any further.
+- Follow the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/coding-standards/).
```
This is a Golang codebase, no WordPress code here.

jeffpaul replied:
@valendesigns any specific coding standards that are being followed then instead of WPCS?

jeffpaul committed:
integrate readme info from old repo

- reorg screenshots + logo into docs/images folder
- add logo + quote to readme header
- reorg intro to its own named section

jeffpaul committed:
add dan and jp to props

jeffpaul committed:
removes mentions of coding standards

jeffpaul committed:
add Code of Conduct

we may want to update the email address here to something tide-specific and potentially outside @xwp.co

jeffpaul committed:
add code of conduct link to contrib guidelines

may want to consider updating to tide-specific email and potentially not on @xwp.co

jeffpaul committed:
add contact us + slack details

jeffpaul committed:
add issue template

jeffpaul committed:
add pr template

valendesigns commented:
I've merged most of these changes into the `api` branch, closing in favor of #42
